src/proof/theory_proof.h

```diff
@@ -234,8 +258,12 @@ class TheoryProof {
    *
    * @param term expression representing term
    * @param os output stream
+   * @param expectedType The type that this is expected to have in a parent node.
+   *                      Null if there are no such requirements.
+   *                      This is useful for requesting type conversions from the theory.
+   *                      e.g. in (5.5 == 4) the right-hand-side should be converted to a real.
    */
-  virtual void printOwnedTerm(Expr term, std::ostream& os, const ProofLetMap& map) = 0;
+  virtual void printOwnedTerm(Expr term, std::ostream& os, const ProofLetMap& map, Type expectedType = Type()) = 0;
```

Comment:
Do you expect to usually use a `Type` or a `TypeNode`?

Reply:
From talking to Andres: `TypeNode` is to `Type` as `Node` is to `Expr`.
I should be using `TypeNode`. I'll update the PR.